tests/Generator/Resolver/Value/Chainable/FixtureMethodCallReferenceResolverTest.php

```diff
@@ -0,0 +1,215 @@
+
+            $dummyProphecy = $this->prophesize(DummyWithGetter::class);
+            $dummyProphecy->getFoo()->willThrow($ex = new \Exception('Inner exception'));
+
+            $valueResolverProphecy = $this->prophesize(ValueResolverInterface::class);
+            $valueResolverProphecy
+                ->resolve($reference, Argument::cetera())
+                ->willReturn(
+                    new ResolvedValueWithFixtureSet(
+                        $instance = $dummyProphecy->reveal(),
+                        $newSet = ResolvedFixtureSetFactory::create(new ParameterBag())
+                    )
+                )
+            ;
+            /** @var ValueResolverInterface $valueResolver */
+            $valueResolver = $valueResolverProphecy->reveal();
+
+            $resolver = new FixtureMethodCallReferenceResolver($valueResolver);
+            $resolver->resolve($value, new FakeFixture(), $set, [], new GenerationContext());
+
+            $this->fail('Expected exception to be thrown.');
+        } catch (UnresolvableValueException $exception) {
+            $this->assertEquals(
+                'Could not resolve value "mutable->getFoo()".',
+                $exception->getMessage()
+            );
+            $this->assertEquals(0, $exception->getCode());
+            $this->assertEquals($ex, $exception->getPrevious());
+        }
+    }
+
+    public function testThrowsAnExceptionIfResolvedReferenceHasNoSuchMethod()
+    {
+        try {
+            $instance = new DummyWithGetter();
+            $value = new FixtureMethodCallValue(
+                $reference = new MutableValue($instance),
+                $functionCall = new FunctionCallValue('getNonExistent')
+            );
+
+            $set = ResolvedFixtureSetFactory::create();
+
+            $valueResolverProphecy = $this->prophesize(ValueResolverInterface::class);
+            $valueResolverProphecy
+                ->resolve(Argument::cetera())
+                ->willReturn(
+                    new ResolvedValueWithFixtureSet(new \stdClass(), $set)
+                )
+            ;
+            /** @var ValueResolverInterface $valueResolver */
+            $valueResolver = $valueResolverProphecy->reveal();
+
+            $resolver = new FixtureMethodCallReferenceResolver($valueResolver);
+            $resolver->resolve(
+                $value,
+                new SimpleFixture('dummy', 'Dummy', SpecificationBagFactory::create()),
+                $set,
+                [],
+                new GenerationContext()
+            );
+
+            $this->fail('Expected exception to be thrown.');
+        } catch (NoSuchMethodException $exception) {
+            $this->assertEquals(
+                'Could not find the method "getNonExistent" of the object "dummy" (class: Dummy).',
+                $exception->getMessage()
+            );
+            $this->assertEquals(0, $exception->getCode());
```

Comment:
there is not previous in that case? (if so I think a `$this->assertNull($exception->getPrevious))` would be appropriate